core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -1280,7 +1286,19 @@ private boolean joinDynamicFilteringExpression(Expression expression, Collection
                 }
                 comparison = (ComparisonExpression) expression;
             }
-            return comparison.getLeft() instanceof SymbolReference && comparison.getRight() instanceof SymbolReference;
+
+            return dynamicFilteringApplicable(comparison, leftSymbols, rightSymbols);
+        }
+
+        // sub expression that uses right source has to be a symbol (left can be an arbitrary expression)
+        private boolean dynamicFilteringApplicable(ComparisonExpression comparison, Collection<Symbol> leftSymbols, Collection<Symbol> rightSymbols)
+        {
+            if (comparison.getRight() instanceof SymbolReference && rightSymbols.contains(Symbol.from(comparison.getRight()))) {
```

Comment:
make it single expressions:
```
return (comparison.getRight() instanceof SymbolReference && rightSymbols.contains(Symbol.from(comparison.getRight())))
  || (comparison.getLeft() instanceof SymbolReference && rightSymbols.contains(Symbol.from(comparison.getLeft()))
```

`joinComparisonExpression` is making sure that side of expressions are using symbols from both sides of join

Reply:
done (but the second condition uses leftSymbols)